src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -4732,6 +4742,43 @@ def get_tags(self) -> Union[Dict[str, str], None]:
         # this parameter does not need validation
         return tags
 
+    def get_defender_config(self) -> Union[Dict[str, str], None]:
+        """Obtain the value of defender.
+
+        :return: string or None
+        """
+        disable_defender = self.raw_param.get("disable_defender")
+        if disable_defender:
+             return {"enabled": False, "logAnalyticsWorkspaceResourceId": ""}
+
+        # read the original value passed by the command
+        enable_defender = self.raw_param.get("enable_defender")
+
+        if not enable_defender:
+            return None
+
+        workspace = ""
+        
+        config_file_path = self.raw_param.get("defender_config")
+        if config_file_path:
+            if not os.path.isfile(config_file_path):
+                raise InvalidArgumentValueError(
+                    "{} is not valid file, or not accessable.".format(
+                        config_file_path
+                    )
+                )
+            defender_config = get_file_json(config_file_path)
+
+            if "logAnalyticsWorkspaceResourceId" in defender_config: 
+                workspace = defender_config["logAnalyticsWorkspaceResourceId"]
+
+        if workspace == "":
+            workspace = ensure_default_log_analytics_workspace_for_monitoring(self.cmd,
```

Comment:
addonconfiguration.py
AzureCloudLocationToOmsRegionCodeMap 

missing the following keys/values:
"australiacentral2": "CBR2",
"canadaeast": "YQ",
"francesouth": "MRS",
"japanwest": "OS",
"jioindiacentral": "JINC",
"jioindiawest": "JINW",
"koreasouth": "PS",
"norwaywest": "NOW",
"southindia": "MA",
"swedencentral": "SEC",
"switzerlandnorth": "CHN",
"westus3": "USW3"

Reply:
Also it won't work on all non-public clouds.
Please add a bug on us / on CLI team to fix it (add relevant regions to the mapping)